src/browser/api_protocol/transport_strategy/ws_strategy.ts

```diff
@@ -44,12 +44,17 @@ export class WebSocketStrategy extends ApiTransportBase<MessagePackage> {
     }
 
     public send(externalConnection: any, payload: any): void {
+        const id = externalConnection.id;
         try {
-            log.writeToLog('info', `sent external-adapter <= ${externalConnection.id} ${JSON.stringify(payload)}`);
+            log.writeToLog('info', `sent external-adapter <= ${id} ${JSON.stringify(payload)}`);
         } catch (err) {
             /* tslint:disable: no-empty */
         }
-        socketServer.send(externalConnection.id, JSON.stringify(payload));
+
+        // Make sure not to send any message to a closed/closing websocket.
+        if (socketServer.isConnectionOpen(id)) {
+            socketServer.send(id, JSON.stringify(payload));
+        }
```

Comment:
Should we maybe log when messages are not being sent, so we know what's being ignored...?

Reply:
I meant to say that we should log that the message is being ignored with some meta-information, but we don't have to log the payload/content of the message

Reply:
what kind of meta info should  we log? It looks to me "id" and "payload" already have all the info.